extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/BeanArchiveIndexBuildItem.java

```diff
@@ -8,6 +8,17 @@
 
 import io.quarkus.builder.item.SimpleBuildItem;
 
+/**
+ * Represent a Jandex {@link IndexView} on the whole deployment that has a complete CDI-related information.
+ *
+ * Compared to {@link io.quarkus.deployment.builditem.CombinedIndexBuildItem} this index can contain additional classes
+ * that were indexed while bean discovery was in progress.
+ *
+ * It also holds information about all programmatically registered beans and all generated bean classes.
```

Comment:
I think we should add some information about when it makes sense to use that one instead of `CombinedIndexBuildItem`.

Reply:
Basically you should use this for anything CDI-related. I will add here something along those lines.